src/MarianInterface.cpp

```diff
@@ -104,34 +99,48 @@ MarianInterface::MarianInterface(QObject *parent)
             emit pendingChanged(true);
 
             try {
-                if (model) {
-                    // Unload marian first (so we can delete loggers after that)
+                if (modelChange) {
+                    // Reconstruct the service because cpu_threads might have changed.
+                    // @TODO: don't recreate Service if cpu_threads didn't change?
+                    marian::bergamot::AsyncService::Config serviceConfig;
+                    serviceConfig.numWorkers = modelChange->settings.cpu_threads;
+                    
+                    // Free up old service first (see https://github.com/browsermt/bergamot-translator/issues/290)
                     service.reset();
 
-                    // We need to manually destroy the loggers, as marian doesn't do
-                    // that but will complain when a new marian::Config tries to 
-                    // initialise loggers with the same name.
-                    spdlog::drop("general");
-                    spdlog::drop("valid");
+                    service = std::make_unique<marian::bergamot::AsyncService>(serviceConfig);
 
-                    service.reset(new marian::bergamot::Service(MakeOptions(model->config_file, model->settings)));
+                    // Initialise a new model. Old model will be released if
+                    // service is done with it, which it is since all translation
+                    // requests are effectively blocking in this thread.
+                    auto modelConfig = makeOptions(modelChange->config_file, modelChange->settings);
+                    model = std::make_shared<marian::bergamot::TranslationModel>(modelConfig, marian::bergamot::MemoryBundle{}, modelChange->settings.cpu_threads);
                 } else if (input) {
-                    if (service) {
-                        auto start = std::chrono::steady_clock::now(); // Time the translation
-                        std::future<int> num_words = std::async(countWords, *input); // @TODO we're doing an "unnecessary" string copy here (necessary because we std::move input into service->translate)
+                    if (model) {
+                        std::future<int> wordCount = std::async(countWords, *input); // @TODO we're doing an "unnecessary" string copy here (necessary because we std::move input into service->translate)
 
                         marian::bergamot::ResponseOptions options;
                         options.alignment = true;
-                        options.alignmentThreshold = 0.1f;
```

Comment:
Is this hardcoded now via soft alignment?

Reply:
Yep, its currently hard-coded in the for-loop, waiting to be replaced by something a bit more dynamic.